src/extensions/ocw.js

```diff
@@ -0,0 +1,111 @@
+const settings = require("../settings.js");
+const parsing = require("./helpers/codeparsing.js");
+
+// Need a slightly different regex because we don't know what OCW IDs will end with
+// If anyone figures that out, we can update this, but it would still be different
+const delim = "[-. ]?";
+const code = "[A-Ha-hJ-Nj-nP-Yp-y0-9]{3}";
+const levelCodeRegexStrict = new RegExp(
+  `^(${code})${delim}(${code})${delim}(${code})$`
+);
+const levelCodeRegex = new RegExp(
+  `(${code})${delim}(${code})${delim}(${code})`
+);
+
+function courseIdValidity(courseIdString) {
+  //
+  let parsed = parsing.courseIdValidity(courseIdString);
+
+  // Make sure it's an OCW code
+  if (parsed.type !== parsing.CodeTypes.OCW) {
+    return { valid: false, makerCode: false };
+  }
+
+  // And then return the parsed results
+  return { valid: parsed.valid, makerCode: parsed.makerCode };
+}
+
+const extractValidCode = (levelCode, lenient = false) => {
+  // TODO: iterate through matches and check if the code is valid for each match and return the first valid one to make this even more lenient
```

Comment:
`ocw-lenient` isn't quite working yet, but this is because I did not implement iterating through all matches.
The tests for that are commented out, but I think the changes can already be merged into develop regardless.
The reason why `smm2-lenient` works better is because of the last digit has to be F, G, or H requirement.

Reply:
Hmm... even iterating through all matches might not work here. We might want to rearchitect the lenient versions a little.

Testing on regex101, it actually only picks up two matches: `please pla` and `vel N41-Q4W`. Dropping space as a deliminator makes it work with that code, though I'm not convinced that's the best solution... I'll open an issue for this, we can discuss it more there.